src/pykeen/models/nmodel/regularizers.py

```diff
@@ -0,0 +1,256 @@
+# -*- coding: utf-8 -*-
+
+"""Regularization in PyKEEN."""
+
+from abc import ABC, abstractmethod
+from typing import Any, ClassVar, Collection, Iterable, List, Mapping, Optional, Sequence, Type, Union
+
+import torch
+from torch import nn
+from torch.nn import functional
+
+from ...utils import get_cls, normalize_string
+from ...regularizers import Regularizer
+
+__all__ = [
+    'NewRegularizer',
+    'LpRegularizer',
+    'NoRegularizer',
+    'CombinedRegularizer',
+    'PowerSumRegularizer',
+    'TransHRegularizer',
+    'collect_regularization_terms',
+    'get_regularizer_cls',
+]
+
+_REGULARIZER_SUFFIX = 'Regularizer'
+
+
+class NewRegularizer(Regularizer):
```

Comment:
The creation of new-style regularizers should probably get split into its own PR as well, since I made a few changes to the base class to facilitate this